bioblend/galaxy/libraries/__init__.py

```diff
@@ -304,6 +335,10 @@ def _do_upload(self, library_id, **keywords):
             payload["roles"] = keywords["roles"]
         if keywords.get("link_data_only", None) and keywords['link_data_only'] != 'copy_files':
             payload["link_data_only"] = 'link_to_files'
+        if keywords.get('tag_using_filenames', False):
+            payload["tag_using_filenames"] = False
+        if keywords.get('preserve_dirs', True):
+            payload["preserve_dirs"] = True
```

Comment:
What about

```python
payload['tag_using_filenames'] = str(keywords.get('tag_using_filenames', True))
payload['preserve_dirs'] = str(keywords.get('preserve_dirs', False))
```

? The casting to string should solve the `MultipartEncoder` encode issue (in https://github.com/galaxyproject/bioblend/blob/master/bioblend/galaxyclient.py#L124 when there are files attached we don't json.dumps() the payload.